pkg/interface/src/views/components/OverlaySigil.tsx

```diff
@@ -1,134 +1,103 @@
+  const [space, setSpace] = useState<OverlaySigilState.space>({
+    top: 'auto',
+    bottom: 'auto'
+  });
+
+  const updateContainerOffset = () => {
+    if (scrollWindow && containerRef && containerRef.current) {
+      const container = containerRef.current;
+      const bottomSpace = scrollWindow
+        ? scrollWindow.clientHeight -
+          (container.getBoundingClientRect().top +
+            OVERLAY_HEIGHT -
+            scrollWindow.getBoundingClientRect().top)
+        : 'auto';
+      const topSpace = scrollWindow
+        ? container.getBoundingClientRect().top -
+          scrollWindow.getBoundingClientRect().top
+        : 0;
+      setSpace({
+        ...space,
+        top: topSpace,
+        bottom: bottomSpace
       });
     }
-  }
-
-  componentWillUnmount() {
-    this.props.scrollWindow?.removeEventListener('scroll', this.updateContainerOffset, true);
-  }
-
-  render() {
-    const {
-      className,
-      ship,
-      contact,
-      color,
-      group,
-      history,
-      api,
-      sigilClass,
-      hideAvatars,
-      pr = 0,
-      pl = 0,
-      ...rest
-    } = this.props;
-
-    const { state } = this;
-
-    const img = (contact && (contact.avatar !== null) && !hideAvatars)
-      ? <BaseImage display='inline-block' src={contact.avatar} height={16} width={16} />
-      : <Sigil
-        ship={ship}
-        size={16}
+    setVisible(true);
+  };
+
+  useEffect(() => {
+    updateContainerOffset();
+    if (scrollWindow)
+      scrollWindow.addEventListener('scroll', updateContainerOffset);
+    return () => {
+      if (scrollWindow)
+        scrollWindow.removeEventListener('scroll', updateContainerOffset, true);
+    };
+  }, []);
+
+  return (
+    <Box
+      cursor='pointer'
+      position='relative'
+      className={className}
+      pr={0}
+      pl={0}
+      ref={containerRef}
+      style={{ visibility: visible ? 'visible' : 'hidden' }}
+    >
+      <ProfileOverlay
+        api={api}
+        bottomSpace={space.bottom}
         color={color}
-        classes={sigilClass}
-        icon
-        padded
-        />;
-
-     return (
-      <Box
-        position='relative'
-        onClick={this.profileShow}
-         ref={this.containerRef}
-         className={className}
-         pr={pr}
-         pl={pl}
-      >
-        {state.clicked && (
-          <ProfileOverlay
-            ship={ship}
-            contact={contact}
-            color={color}
-            topSpace={state.topSpace}
-            bottomSpace={state.bottomSpace}
-            group={group}
-            onDismiss={this.profileHide}
-            history={history}
-             api={api}
-             {...rest}
-          />
-        )}
-        {img}
-      </Box>
-    );
-  }
-}
-
-export default withLocalState(OverlaySigil, ['hideAvatars']);
+        contact={contact}
+        group={group}
+        history={history}
+        onDismiss={() => onDismiss()}
```

Comment:
any reason for the unnecessary arrow functioN?

Reply:
None, will fix

Reply:
My mistake -- this is calling the parent function that's passed in, which sets that `showOverlay` flag one component up. Is there a better way to handle this?

Reply:
Just do `onDismiss={onDismiss}`

Reply:
Actually you could just remove `api`, `color`, `contact`, `group`, and `history` from the destructuring of props and they'd get passed in with `...rest`